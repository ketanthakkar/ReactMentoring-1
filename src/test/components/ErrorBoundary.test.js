import React from 'react';
import { shallow } from 'enzyme';

import ErrorBoundary from 'components/ErrorBoundary';

describe('ErrorBoundary', () => {
    const errorBoundary = shallow(<ErrorBoundary />);

    it('should be like snapshot', () => {
        errorBoundary.setState({error: 'error', errorInfo: {}});
        expect(errorBoundary).toMatchSnapshot();
    });
});