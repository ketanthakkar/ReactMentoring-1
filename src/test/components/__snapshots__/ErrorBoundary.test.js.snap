// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should be like snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorBoundary />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Something went wrong.
        </h2>,
        <details
          style={
            Object {
              "whiteSpace": "pre-wrap",
            }
          }
        >
          error
          <br />
        </details>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Something went wrong.",
        },
        "ref": null,
        "rendered": "Something went wrong.",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "error",
            <br />,
            undefined,
          ],
          "style": Object {
            "whiteSpace": "pre-wrap",
          },
        },
        "ref": null,
        "rendered": Array [
          "error",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          undefined,
        ],
        "type": "details",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Something went wrong.
          </h2>,
          <details
            style={
              Object {
                "whiteSpace": "pre-wrap",
              }
            }
          >
            error
            <br />
          </details>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something went wrong.",
          },
          "ref": null,
          "rendered": "Something went wrong.",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "error",
              <br />,
              undefined,
            ],
            "style": Object {
              "whiteSpace": "pre-wrap",
            },
          },
          "ref": null,
          "rendered": Array [
            "error",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            undefined,
          ],
          "type": "details",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
